pupil_src/shared_modules/calibration_routines/data_processing.py

```diff
@@ -0,0 +1,200 @@
+
+    if len(matched_pupil0_data) > len(matched_pupil1_data):
+        matched_monocular_data = matched_pupil0_data
+    else:
+        matched_monocular_data = matched_pupil1_data
+
+    logger.info(
+        "Collected {} monocular calibration data.".format(len(matched_monocular_data))
+    )
+    logger.info(
+        "Collected {} binocular calibration data.".format(len(matched_binocular_data))
+    )
+    return (
+        matched_binocular_data,
+        matched_monocular_data,
+        matched_pupil0_data,
+        matched_pupil1_data,
+        pupil0,
+        pupil1,
+    )
+
+
+def closest_matches_binocular(ref_pts, pupil_pts, max_dispersion=1 / 15.0):
+    """
+    get pupil positions closest in time to ref points.
+    return list of dict with matching ref, pupil0 and pupil1 data triplets.
+    """
+    pupil0 = [p for p in pupil_pts if p["id"] == 0]
+    pupil1 = [p for p in pupil_pts if p["id"] == 1]
+
+    pupil0_ts = np.array([p["timestamp"] for p in pupil0])
+    pupil1_ts = np.array([p["timestamp"] for p in pupil1])
+
+    matched = []
+
+    if pupil0 and pupil1:
+        for r in ref_pts:
+
+            closest_p0_idx = find_nearest_idx(pupil0_ts, r["timestamp"])
+            closest_p0 = pupil0[closest_p0_idx]
+            closest_p1_idx = find_nearest_idx(pupil1_ts, r["timestamp"])
+            closest_p1 = pupil1[closest_p1_idx]
+
+            dispersion = max(
+                closest_p0["timestamp"], closest_p1["timestamp"], r["timestamp"]
+            ) - min(closest_p0["timestamp"], closest_p1["timestamp"], r["timestamp"])
+            if dispersion < max_dispersion:
+                matched.append({"ref": r, "pupil": closest_p0, "pupil1": closest_p1})
+            else:
+                logger.debug(
+                    "Binocular match rejected due to time dispersion criterion"
+                )
+
+    return matched
+
+
+def closest_matches_monocular(ref_pts, pupil_pts, max_dispersion=1 / 15.0):
+    """
+    get pupil positions closest in time to ref points.
+    return list of dict with matching ref and pupil datum.
+
+    if your data is binocular use:
+    pupil0 = [p for p in pupil_pts if p['id']==0]
+    pupil1 = [p for p in pupil_pts if p['id']==1]
+    to get the desired eye and pass it as pupil_pts
+    """
+
+    pupil0 = pupil_pts
+    pupil0_ts = np.array([p["timestamp"] for p in pupil0])
+
+    matched = []
+
+    if pupil0:
+        for r in ref_pts:
+            closest_p0_idx = find_nearest_idx(pupil0_ts, r["timestamp"])
+            closest_p0 = pupil0[closest_p0_idx]
+            dispersion = np.abs(closest_p0["timestamp"] - r["timestamp"])
+            if dispersion < max_dispersion:
+                matched.append({"ref": r, "pupil": closest_p0})
+            else:
+                logger.debug(
+                    "Monocular match rejected due to time dispersion criterion"
+                )
+
+    return matched
+
+
+def preprocess_2d_data_monocular(matched_data):
```

Comment:
`preprocess` is a strong/very general word for what happens here. Maybe something like `extract_calibration_data` or `extract_relevant_fields` or so would be more appropriate.